### ISSUE TITLE
License, code, alternate packages ... mixed bag ;)

### ISSUE THREAD

**Issue by dudeskeeroo** [brianpetro/smart-connect#1]
Hi,

Firstly, thank you for the great plugin.

I stumbled on this project but haven't been able to locate the license, or source code, or a privacy policy. Everything seems quite opaque, which makes me reluctant to try it out.

Also, for users on non-Debian systems, we can't install (no code, no alternate installers etc.)

Just thought I'd feed this info back, as a user of SC of Obsidian.

Cheers.

**Comment by brianpetro** [comment]
@dudeskeeroo much of the repository is available in JS Brains.

While free and a vehicle for developing the open-source [JS Brains](https://github.com/brianpetro/jsbrains), Smart Connect core will continue to be a proprietary arm of Smart Connections. I imagine this will be similar to how Obsidian is free, still prioritizes privacy and security, but maintains their core code as private.

Are you interested in a non-debian package for Smart Connect? I can compile one, I just haven't been using Linux for a while, so need early adopters to help with testing, and that could be you 😊

🌴

**Comment by dudeskeeroo** [comment]
Thanks for your quick reply!

The way most Linux projects seem to be distributing these days is using [Flatpak](https://flatpak.org/). Flatpaks are distro independent, contain all the dependent libs inside the pak, and allow sandboxing. Like everything, there are pros/cons, tradeoffs etc.

I'm using Fedora and some of the apps that come as Flatpaks, as an example, are Firefox, Obsidian, Discord, Inkkscape, Gimp, Meld, VSCodium, and more.

You may need a Linux VM or a docker image to do the build. I've never created a flatpak, so this is about as much as I really know of the ins-and-outs of it all, I'm afraid.

**Comment by dudeskeeroo** [comment]
Hi!

I downloaded the Appimage and fired it up without any problems. I'm not currently a supporter and don't subscribe to ChatGPT+ so my usage was very limited. I'm not sure if the README mentions the requirement of being a supporter ... I assumed the requirement _license key_ was my OpenAI API key, but that was clearly my misreading.

The UI came up perfectly fine and I could interact with it. There didn't seem to be any issues with running the Appimage.

I've included the terminal log output in case you find it interesting. It looks like there are errors in the output, but the app didn't seem to be buggy at all. I exited the app cleanly before copying the output below:

<details>
  <summary>Terminal output</summary>

```
$ /tmp/Smart-Connect-Setup-linux-x86_64.AppImage         
[Main.create 54:13] App event: Ready
[Main.init 61:15] Starting power save blocker...
[Main.init 64:13] App event: Ready
[Main.open 431:27] Using bundled preload
Checking for update
[Main.update_main_status 212:13] Checking for update...
[Main.register_auto_update 355:13] Auto-updater registered
DRM kernel driver 'nvidia-drm' in use. NVK requires nouveau.
Generated new staging user ID: ea5724d6-5444-550f-bc0e-8ed2874c8eb4
[Main.load_config 76:15] Loading settings...
[Main.update_main_status 212:13] v1.1.50
Update for version 1.1.50 is not available (latest version: 1.1.50, downgrade is disallowed).
[Main.update_main_status 212:13] Update not available.
checkForUpdatesAndNotify called, downloadPromise is null
[Main.update_main_status 212:13] v1.1.50
Warning: loader_get_json: Failed to open JSON file broadcom_icd.x86_64.json
Warning: loader_get_json: Failed to open JSON file freedreno_icd.x86_64.json
Warning: loader_get_json: Failed to open JSON file intel_hasvk_icd.x86_64.json
Warning: /usr/lib64/libvulkan_intel.so: cannot open shared object file: Permission denied
Warning: loader_icd_scan: Failed loading library associated with ICD JSON /usr/lib64/libvulkan_intel.so. Ignoring this JSON
Warning: loader_get_json: Failed to open JSON file lvp_icd.x86_64.json
Warning: loader_get_json: Failed to open JSON file nouveau_icd.x86_64.json
Warning: loader_get_json: Failed to open JSON file panfrost_icd.x86_64.json
Warning: loader_get_json: Failed to open JSON file powervr_mesa_icd.x86_64.json
Warning: /usr/lib64/libvulkan_radeon.so: cannot open shared object file: Permission denied
Warning: loader_icd_scan: Failed loading library associated with ICD JSON /usr/lib64/libvulkan_radeon.so. Ignoring this JSON
Warning: loader_get_json: Failed to open JSON file broadcom_icd.i686.json
Warning: loader_get_json: Failed to open JSON file freedreno_icd.i686.json
Warning: loader_get_json: Failed to open JSON file intel_hasvk_icd.i686.json
Warning: loader_get_json: Failed to open JSON file intel_icd.i686.json
Warning: loader_get_json: Failed to open JSON file lvp_icd.i686.json
Warning: loader_get_json: Failed to open JSON file nouveau_icd.i686.json
Warning: loader_get_json: Failed to open JSON file panfrost_icd.i686.json
Warning: loader_get_json: Failed to open JSON file powervr_mesa_icd.i686.json
Warning: loader_get_json: Failed to open JSON file radeon_icd.i686.json
Warning: loader_scanned_icd_add: Could not get 'vkCreateInstance' via 'vk_icdGetInstanceProcAddr' for ICD libGLX_nvidia.so.0
Warning: vkCreateInstance: Found no drivers!
Warning: vkCreateInstance failed with VK_ERROR_INCOMPATIBLE_DRIVER
    at CheckVkSuccessImpl (../../third_party/dawn/src/dawn/native/vulkan/VulkanError.cpp:101)
    at CreateVkInstance (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:493)
    at Initialize (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:379)
    at Create (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:301)
    at operator() (../../third_party/dawn/src/dawn/native/vulkan/BackendVk.cpp:556)

[184956:0501/101920.787714:ERROR:CONSOLE(1)] "Uncaught (in promise) TypeError: Failed to fetch", source: devtools://devtools/bundled/panels/elements/elements.js (1)
[184956:0501/101924.407282:ERROR:CONSOLE(0)] "ReferenceError: dragEvent is not defined", source:  (0)
```
</details>

I hope this helps. Best of luck with the product. ;)

**Comment by brianpetro** [comment]
Thanks for the follow up. Those errors are probably ok unless you notice any obvious issues, then let me know.

This software is technically unreleased, and it's hard for me to resist just divulging more... but there will be more coming soon 😉 

🌴
